test/integration/router_test.go

```diff
@@ -494,17 +525,26 @@ func TestRouterPathSpecificity(t *testing.T) {
 				To: kapi.ObjectReference{
 					Name: "altService",
 				},
+				routeTLS: &routeapi.TLSConfig{
+					Termination:                   routeapi.TLSTerminationEdge,
+					Certificate:                   tr.ExampleCert,
+					Key:                           tr.ExampleKey,
+					CACertificate:                 tr.ExampleCACert,
+					InsecureEdgeTerminationPolicy: routeapi.InsecureEdgeTerminationPolicyAllow,
+				},
 			},
 		},
 	}
 	fakeMasterAndPod.EndpointChannel <- eventString(endpointEvent)
 	fakeMasterAndPod.RouteChannel <- eventString(routeEvent)
 
-	if err := waitForRoute(routeTestAddress, "www.example.com", "http", nil, tr.HelloPodPath); err != nil {
-		t.Fatalf("unexpected response: %q", err)
-	}
-	if err := waitForRoute(routeTestAddress, "www.example.com:80", "http", nil, tr.HelloPodPath); err != nil {
-		t.Fatalf("unexpected response: %q", err)
+	for _, proto := range protocols {
+		if err := waitForRoute(routeTestAddress, "www.example.com", proto.name, nil, tr.HelloPodPath); err != nil {
+			t.Fatalf("unexpected response: %q", err)
+		}
+		if err := waitForRoute(routeTestAddress, "www.example.com:" + proto.port, proto.name, nil, tr.HelloPodPath); err != nil {
+			t.Fatalf("unexpected response: %q", err)
```

Comment:
same as previous comment re: adding `protocol` (`proto.name`)  information to the failure message.